File: hopl_app/lib/models/shop.dart
import 'package:hopl_app/models/shopItem.dart';

class Shop {
  final String shopName;
  final String description;
  final String shopId;
  final String imgUrl;
  final int rating;
  final String category;
  final List<ShopItem> items;

  Shop(
      {required this.shopName,
      required this.description,
      required this.shopId,
      required this.rating,
      required this.category,
      required this.items,
      required this.imgUrl});
}

File: hopl_web/Server/models/Shop.ts
import mongoose from "mongoose";

const shopSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  description: String,
  shopKeeperId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
    required: true,
  },
  address: {
    type: String,
    required: true,
  },
  rating: {
    type: Number,
  },
  longitude: {
    type: String,
    required: true,
  },
  latitude: {
    type: String,
    required: true,
  },
  category: String,
  imgUrl: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Image",
  },
  products: [
    { type: mongoose.Schema.Types.ObjectId, ref: "Product", default: [] },
  ],
});

export default mongoose.models.Shop || mongoose.model("Shop", shopSchema);
